### PR TITLE
on-chain poseidon benches

### PR DESCRIPTION
**Pull request by fbielejec**
# Description

Benchmark poseidon-using relation on-chain. This contract verifies a proof of a preimage relation (by means of runtime call) as part of a reveal tx. This relation included one evaluation of 1:1 Poseseidon hash function. 
Results:
```bash
    Success! Gas required estimated at Weight(ref_time: 84398140337, proof_size: 131072)
      Events
       Event Balances ➜ Withdraw
         who: 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
         amount: 85.210685688mDZERO
       Event BabyLiminal ➜ VerificationSucceeded
       Event Contracts ➜ Called
         caller: 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
         contract: 5EFktxrdBqjJqwxDheMNZr48jBYApVVyPpzWax2kgJMbUu9h
       Event Balances ➜ ReserveRepatriated
         from: 5EFktxrdBqjJqwxDheMNZr48jBYApVVyPpzWax2kgJMbUu9h
         to: 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
         amount: 64mDZERO
         destination_status: Free
       Event Balances ➜ Deposit
         who: 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
         amount: 37.53137174mDZERO
       Event Balances ➜ Deposit
         who: 5EYCAe5fg5WiYGVNH6QpCFnu55Hzv9MwtjFHdQCx8EaSQTm2
         amount: 47.679313948mDZERO
       Event Treasury ➜ Deposit
         value: 47679313948
       Event TransactionPayment ➜ TransactionFeePaid
         who: 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
         **actual_fee: 47.679313948mDZERO**
         tip: 0DZERO
       Event System ➜ ExtrinsicSuccess
         dispatch_info: DispatchInfo { weight: Weight { ref_time: 47679313597, proof_size: 14919 }, class: Normal, pays_fee: Yes }

```

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have created new documentation

### CODE REVIEW

File: contracts/preimage/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+edition = "2021"
+
+[dependencies]
+ark-bls12-381 = { version = "^0.3.0" }
+ark-ff = { version = "^0.3.0", default-features = false }
+ark-serialize = { version = "^0.3.0", default-features = false }
+ink = { git = "https://github.com/paritytech/ink", rev = "4655a8b4413cb50cbc38d1b7c173ad426ab06cde", default-features = false }
+scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
+scale-info = { version = "2.3", default-features = false, features = ["derive"], optional = true }
+snarcos-extension = { git = "https://github.com/Cardinal-Cryptography/aleph-node", branch = "liminal/migrate-to-ink4", package = "snarcos-extension", default-features = false, features = ["ink-as-dependency"] }
```
Comment (Marcin-Radecki) [inline contracts/preimage/Cargo.toml:14]: I cannot see branch `liminal/migrate-to-ink4` in `aleph-node`, do I miss sth

**Review by Marcin-Radecki — APPROVED** [review]
Please note I have very limited knowledge in the topic :)

File: contracts/preimage/Cargo.toml (hunk 1 above)
Comment (fbielejec) [reply contracts/preimage/Cargo.toml:14]: Good catch - this is now in the upstream branch and re-named to `baby-liminal-extension`. Fixed

**Comment by fbielejec** [comment]
abandoned since Poseidon is in pallet now and the calls are benchamarked there

**Pull request closed without merging**